media-sound/clementine/patches/clementine-1.4.1~git.patchset

```diff
@@ -439,3 +439,45 @@ index fbe2693..fa5a915 100644
 -- 
 2.48.1
 
+
+From 6780ba10b1b690ae9483f20bbb3b396c8cb73307 Mon Sep 17 00:00:00 2001
+From: Ken Mays <kmays2000@gmail.com>
+Date: Fri, 2 May 2025 05:13:42 +0004
+Subject: Enable visualization on Haiku
+
+
+diff --git a/3rdparty/libprojectm/Renderer/FBO.hpp b/3rdparty/libprojectm/Renderer/FBO.hpp
+index e79954d..efbe620 100644
+--- a/3rdparty/libprojectm/Renderer/FBO.hpp
++++ b/3rdparty/libprojectm/Renderer/FBO.hpp
+@@ -50,7 +50,6 @@
+ #include <GLES/gl.h>
+ #else
+ #include <GL/gl.h>
+-#include <GL/glx.h>
+ #endif
+ #endif
+ 
+diff --git a/3rdparty/libprojectm/Renderer/SOIL/SOIL.c b/3rdparty/libprojectm/Renderer/SOIL/SOIL.c
+index 2403db2..0ee6cfe 100644
+--- a/3rdparty/libprojectm/Renderer/SOIL/SOIL.c
++++ b/3rdparty/libprojectm/Renderer/SOIL/SOIL.c
+@@ -27,7 +27,6 @@
+ 	#define APIENTRY
+ #else
+ 	#include <GL/gl.h>
+-	#include <GL/glx.h>
+ #endif
+ 
+ #include "SOIL.h"
+@@ -2012,7 +2011,6 @@ int query_DXT_capability( void )
+ 				CFRelease( bundle );
+ 			#else
+ 				ext_addr = (P_SOIL_GLCOMPRESSEDTEXIMAGE2DPROC)
+-						glXGetProcAddressARB

```

Comment:
How is this supposed to work? This removes the function name from a call. You are now effectively casting the argument string literal to a function pointer, which is surely wrong.

I don't know what the equivalent of `glXGetProcAddressARB` in HGL (the Haiku OpenGL kit) is, but I guess this must be used here.

Reply:
The actual function is invalid for non-X and non_Windows. Official UNIX is X-based.
Basically, ext_addr wants the address or a non-NULL/NULL value to work/not work... etc, etc.....

Reply:
Is this `ext_addr` not actually called anywhere? Anyway, casting a string literal to a function pointer is always wrong (which the current version of the patchset still does).